sql/sql_audit.cc

```diff
@@ -451,7 +451,48 @@ int mysql_audit_notify(THD *thd, mysql_event_general_subclass_t subclass,
   event.general_host= sctx->host();
   event.general_external_user= sctx->external_user();
   event.general_rows= thd->get_stmt_da()->current_row_for_condition();
-  event.general_sql_command= sql_statement_names[thd->lex->sql_command];
+  if(thd->lex->sql_command == SQLCOM_END && msg_len > 0 )
+  {
+    if(!strcmp(msg,"Prepare"))
+    {
+      event.general_sql_command.str= "stmt_prepare";
```

Comment:
BTW, here, you could do `event.general_sql_command = C_STRING_WITH_LEN("...")`

Reply:
I had tried with C_STRING_WITH_LEN("...") ,But I was getting error " error: extended initializer lists only available with ‘-std=c++11’ or ‘-std=gnu++11’ [-Werror]" 
This is warning and warnings are treated as error , so , it's not building with  this change.

May be because of this sql_statement_names is built using 'str' and 'len' members explicitly : 
./sql/mysqld.cc:      sql_statement_names[com_index].str= var->name;
./sql/mysqld.cc:      sql_statement_names[com_index].length= strlen(var->name);
 and 

event.general_command.str= msg;
event.general_command.length= msg_len;

Reply:
Please try like this:  = { C_STRING_WITH_LEN("Host") }. (With curly brackets)

Reply:
I was trying with curly brackets only.
event.general_sql_command = { C_STRING_WITH_LEN("Host") };

is not compiling .

Reply:
I think inside makefile , CXX flags must be set as : -std=c++03

Reply:
I have verified this with test program 
g++ test.cpp -o test -std=c++03
test.cpp: In function ‘int main()’:
test.cpp:19:33: warning: extended initializer lists only available with ‘-std=c++11’ or ‘-std=gnu++11’
   19 |  t.test = {"test",strlen("test")};

g++ test.cpp -o test -std=c++11 :  No warning

Reply:
How does it work in other places. It is used a lot in 5.7 codebase. Something else is the reason? :thinking:

Reply:
Inside file : sql_audit.cc , I am not able to use any of c++ 11 features.I am not sure why

Reply:
AFAIR, c+11 is from 8.0 code base only. C_STRING_WITH_LEN has nothing do with c+11 feature? :thinking:

Reply:
> AFAIR, c+11 is from 8.0 code base only. C_STRING_WITH_LEN has nothing do with c+11 feature?

Yeah right, I was trying to use auto and lamda's

Reply:
> also compare compilation flags of sql_audit.cc with other source files? (use VERBOSE=1 make)

With VERBOSE=1 , I can see sql_audi.cc is compiling with -std=gnu++03 option

Reply:
ok, did you avoid lambdas, autos and C+11 features in 5.7? do you see still same error?

Reply:
I have figured out issue , with -std=c++03
You can initialise a struct with extended initializer lists , but you can't assign .
That means event.general_sql_command = { C_STRING_WITH_LEN(msg)} ; doesn't work 
But, MYSQL_LEX_CSTRING command_class = { C_STRING_WITH_LEN(msg)} ; works.